src/miner.cpp

```diff
@@ -91,6 +91,11 @@ void UpdateTime(CBlockHeader* pblock, const Consensus::Params& consensusParams,
 
 CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
 {
+    LOCK(cs_main);
+    // Are we in sync enough to even try creating a block? Otherwise TestBlockValidity will fail.
+    if (chainActive.Height() < Checkpoints::GetTotalBlocksEstimate())
```

Comment:
I've reproduced the issue to verify: the underlying problem appears to be that the `std::runtime_exception` raised by CreateNewBlock in case `TestBlockValidity` fails is not caught by the internal miner. The other uses of this function in RPC are protected.
```
terminate called after throwing an instance of 'std::runtime_error'
  what():  CreateNewBlock() : TestBlockValidity failed
```
Adding an extra check here at the beginning is redundant - you'd add a check to avoid raising an exception later in the same funcion :) better to change the calling code in `BitcoinMiner` to catch and report this exception.